src/libraries/System.Text.RegularExpressions/gen/DiagnosticDescriptors.cs

```diff
@@ -46,25 +46,16 @@ internal static class DiagnosticDescriptors
             isEnabledByDefault: true,
             customTags: WellKnownDiagnosticTags.NotConfigurable);
 
-        public static DiagnosticDescriptor InvalidLangVersion { get; } = new DiagnosticDescriptor(
-            id: "SYSLIB1044",
-            title: new LocalizableResourceString(nameof(SR.InvalidRegexGeneratorAttributeTitle), SR.ResourceManager, typeof(FxResources.System.Text.RegularExpressions.Generator.SR)),
-            messageFormat: new LocalizableResourceString(nameof(SR.InvalidLangVersionMessage), SR.ResourceManager, typeof(FxResources.System.Text.RegularExpressions.Generator.SR)),
-            category: Category,
-            DiagnosticSeverity.Error,
-            isEnabledByDefault: true,
-            customTags: WellKnownDiagnosticTags.NotConfigurable);
-
         public static DiagnosticDescriptor LimitedSourceGeneration { get; } = new DiagnosticDescriptor(
```

Comment:
I suppose we won't want to change this, but did we consider at all turning this into a warning? I know that is not ideal, but I'm just afraid how many people will entirely miss the fact that they are not actually using the source generator because of their language version, while before this they would know right away as they would get compiler errors when the partial method didn't get filled.

Reply:
From my perspective, a warning is something that a user must take action on because the thing is likely doing something bad (for folks that enable warnings-as-errors it'll fail the compilation).  What would the user's action be here?  If they've explicitly set the LangVersion to be an older C# version, it seems unlikely this alone would motivate upgrading that when presumably there were reasons it was set that way in the first place.  Obviously there could be exceptions, like it's a stale LangVersion setting and they just didn't realize it, but personally I don't think the regex generator should be in that game.  And if someone has to change the code in order avoid the warning, they'd be changing it to effectively the same thing the generator is producing for them in this case.  In short, I'm concerned it'd just be unnecessary noise.  But we could certainly have the conversation if you think it should be a warning.

Reply:
Ok, that's fair. I guess in my mind it was more like:
- Info: Some relevant info that the user may care about. No need for them to do anything.
- Warn: There is something that user **should** know about. They may likely want to do something.
- Error: The user must take action on or specifically suppress.

In my mind this fell in between the first two categories. My feeling towards Warning is that we are not doing what the user is intending/asking for. But my feeling towards Info is that functionality-wise things should still just work, it will just be performance differences from what they might have expected.

Reply:
> If they've explicitly set the LangVersion to be an older C# version, it seems unlikely this alone would motivate upgrading that when presumably there were reasons it was set that way in the first place

I may be wrong, but I think it will be extremely rare for someone to explicitly pin down an older language version on purpose. My hunch is that most people that will be landing on this will be folks that have apps that were targeting older TFMs (like net5.0, etc.) and they had pinned-up the language version to use a newer version, but later updated the TFM to 7.0 without removing the specific pinning of the language. That is kind of why my thoughts were that it would be good for them to be very aware of this, so they can remove the pinning.

Reply:
If that's the reason and there's nothing else preventing them from removing it, I agree.

But what if that's not the reason?

Reply:
All that said, these arguments are only applicable to the case that LangVersion < 11, since I definitely don't think that warning for cases where the pattern isn't supported by the source generator is appropriate, since for those cases, the user will very likely not really have a choice as that is the pattern they need. 

Anyway, we don't need to decide on this and I don't want to block your PR just for this, so I'm fine with having this conversation offline.

Reply:
> But what if that's not the reason?

I guess there is no way to know 😆, that is just what I believe would be the most common case. I've heard Jared make the same hypothesis around people that use an older version of Roslyn than what ships in the SDK.